marlowe-playground-client/src/Simulation.purs

```diff
@@ -389,12 +343,76 @@ render state =
   where
   showRightPanel = state ^. _showRightPanel
 
-  isBlocklyEnabled = view (_marloweState <<< _Head <<< _editorErrors <<< to Array.null) state
-
   demoScriptLink key =
     li [ state ^. _activeDemo <<< activeClasses (eq key) ]
       [ a [ onClick $ const $ Just $ LoadScript key ] [ text key ] ]
 
+otherActions :: forall p. State -> HTML p Action
+otherActions state =
+  div [ classes [ ClassName "group" ] ]
+    ( [ editorOptions state
+      , sendToBlocklyButton state
+      ]
+        <> ( if state ^. (_source <<< to (eq Haskell)) then
```

Comment:
More from the, "I don't know if this is better, but it's interesting so I'm saying it," file:

```purescript
is (_source <<< only Haskell) state
```

Reply:
Having said that, shouldn't all these `if/then/else`s be replaced with a single, exhaustive `case`?

Reply:
`only` ! nice